src/facilities/dto/create-facility.dto.ts

```diff
@@ -1 +1,40 @@
-export class CreateFacilityDto {}
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, IsString } from 'class-validator';
+import { CreateFacilityStatus } from '../create-facility-status.enum';
+
+export class CreateFacilityDto {
+  @ApiProperty({ type: 'number' })
+  @IsNotEmpty()
+  typeid: number;
+
+  @ApiProperty({ type: 'string' })
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+
+  @ApiProperty({ example: 'megha arcade,power house road' })
+  @IsNotEmpty()
+  address: any;
+
+  @ApiProperty({ type: 'number' })
+  @IsNotEmpty()
+  pincode: number;
+
+  @ApiProperty({ type: 'string' })
+  description: string;
+
+  @ApiProperty({ type: 'string' })
+  latitude: string;
+
+  @ApiProperty({ type: 'string' })
+  longitude: string;
+
+  @ApiProperty({ type: 'string' })
+  contact: string;
```

Comment:
What is going to store here can you please provide a sample input

Reply:
{
  "typeid": 1,
  "name": "theatre",
  "address": "megha arcade,power house road",
  "pincode": 686578,
  "description": "string",
  "latitude": "string",
  "longitude": "string",
  "contact": "string",
  "imageUrl": "http://www.google.com/arihant.png",
  "status": "Status:OPEN,CLOSE,IN_ACTIVE"
}